rust/src/measurements/laplace/float/make_scalar_float_laplace_cks20.tex

```diff
@@ -0,0 +1,115 @@
+    
+\begin{tcolorbox}
+    The proof assumes the following lemma.
+    \begin{lemma}
+        \texttt{get\_discretization\_consts}, \texttt{sample\_discrete\_laplace\_Z2k} and \texttt{laplace\_map} each satisfy their postcondition.
+    \end{lemma}
+\end{tcolorbox}
+
+This mechanism can be thought of as a stable transformation from a floating-point number to a rational number,
+followed by a mechanism that adds noise to the rational number.
+
+\subsection{Rounding Transformation}
+
+The transformation rounds the input to the nearest rational number where the denominator is no greater than $2^k$.
+By the postcondition of \texttt{get\_discretization\_consts}, this rounding changes the input by at most \texttt{relaxation}.
+
+\begin{align*}
+    &\max_{x \sim x'} d_{Abs}(round(x), round(x')) \\
+    =& \max_{x \sim x'} \left| round(x) - round(x') \right| \\
+    \leq& \max_{x \sim x'} \left| x - x' \right| + \texttt{relaxation} &&\text{by postcondition of \texttt{get\_discretization\_consts}} \\
+    \leq& \din + \texttt{relaxation} &&\text{by definition of absolute distance}
+\end{align*}
+
+Therefore the rounding transformation is $\din + \texttt{relaxation}$-close under the absolute distance.
+
+\subsection{Noise Measurement}
+\texttt{sample\_discrete\_laplace\_Z2k} can only fail due to lack of system entropy. 
+This is usually related to the computer's physical environment and not the dataset. 
+The rest of this proof is conditioned on the assumption that this function does not raise an exception. 
+
+Let $x$ and $x'$ be datasets that are \texttt{d\_in}-close with respect to \texttt{input\_metric}.
+Here, the metric is \texttt{AbsoluteDistance<T>}.
+By the postcondition of \texttt{sample\_discrete\_laplace\_Z2k},
+the output of the function follows the Discrete Laplace Distribution with scale \texttt{scale}.
+
+\begin{align*}
+    & \max_{x \sim x'} D_{\infty}(M(x), M(x'))  \\
+    =& \max_{x \sim x'} \max_{z \in supp(M(\cdot))} \ln\left(\frac{\Pr\left[M(x) = z\right]}{\Pr\left[M(x') = z \right]}\right)
+        &&\text{substitute } D_{\infty}\\
+    =& \max_{x \sim x'} \max_{z \in \mathbb{Z}} \ln\left(\frac{\Pr\left[\mathrm{DLap}(x, b) = z \right]}{\Pr\left[\mathrm{DLap}(x', b) = z\right]}\right)
+        &&\text{where } b \text{ is the noise scale} \\
+    =& \max_{x \sim x'} \max_{z \in \mathbb{Z}} \ln\left(\frac{
+    \frac{\exp^{1/b} - 1}{\exp^{1/b} + 1} \exp \left( -\frac{|x - z|}{b} \right)
+    }{
+        \frac{\exp^{1/b} - 1}{\exp^{1/b} + 1} \exp \left( -\frac{|x' - z|}{b} \right)
+    }\right) 
+        &&\text{use pdf of Discrete Laplace} \\
+    =& \max_{x \sim x'} \max_{z \in \mathbb{Z}} \ln\left(\frac{
+    \exp \left( -\frac{|x - z|}{b} \right)
+    }{
+        \exp \left( -\frac{|x' - z|}{b} \right)
+    }\right) \\
+    =& \max_{x \sim x'} \max_{z \in \mathbb{Z}}\frac{|x' - z| - |x - z|}{b}&& \text{exp and ln cancel} \\
+    \leq& \frac{\max_{x \sim x'} |x - x'|}{b} &&\text{by reverse triangle inequality} \\
```

Comment:
Might want to remove the "reverse", since that's a different concept.

Reply:
I think this is, actually, the reverse triangle inequality!